### PR TITLE
Homework 2.0

### PR DESCRIPTION
**Pull request by maxutor**

### CODE REVIEW

File: homework-2.0/task7.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        @stack.push(chr)
+      elsif closer?(chr)
+        if closes?(@stack.last, chr)
+          @stack.pop
+        else
+          return false
+        end
+      end
+    end
+    return true if @stack.empty?
```
Comment (slenderock) [inline homework-2.0/task7.rb]: here if you have `@stack.empty?` true - you return true, if it's false you return false? 

I'd rather avoid such a conditions and use just @stack.empty as return value

File: homework-2.0/task7.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  def opener?(chr)
+    BRACKETS.keys.include?(chr)
+  end
+
+  def closer?(chr)
+    BRACKETS.values.include?(chr)
+  end
+
+  def closes?(opener, closer)
+    return false unless BRACKETS[opener] == closer
```
Comment (slenderock) [inline homework-2.0/task7.rb]: > here if you have @stack.empty? true - you return true, if it's false you return false?

> I'd rather aboid such a conditions and use just @stack.empty as return value


same here

**Review by slenderock — CHANGES_REQUESTED** [review]

File: homework-2.0/task3.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  def filter
+    @arr.select { |item| (to_numeric(item) % 2).zero? }
+  end
+
+  private
+
+  def to_numeric(thing)
+    return thing.to_s.to_i if thing.to_s == thing.to_s.to_i.to_s
```
Comment (slenderock) [inline homework-2.0/task3.rb]: `to_s.to_i` doesn't look too good. Mayby consider other solution?

Please let me know if you need any help

File: homework-2.0/task3.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    return thing.to_s.to_f if thing.to_s == thing.to_s.to_f.to_s
```
Comment (slenderock) [inline homework-2.0/task3.rb]: avoid redundant `return` for the last line. Ruby does it by defaut

File: homework-2.0/mario_game.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  def change_speed_enemies(new_speed_enemies)
+    @speed_enemies = new_speed_enemies
+  end
+
+  def change_counf_of_enemies(new_count_of_enemies)
+    @enemies = new_count_of_enemies
+  end
+end
+
+m = MarioGame.new('forest1', 10, 115, 1)
```
Comment (slenderock) [inline homework-2.0/mario_game.rb:38]: It's better to use meaningful variable names. Doesn't required for now. Note for future

File: homework-2.0/task3.rb (hunk 3 above)
Comment (slenderock) [reply homework-2.0/task3.rb]: As an option you can use `regex`

**Review by slenderock — CHANGES_REQUESTED** [review]

File: homework-2.0/task3.rb (hunk 3 above)
Comment (maxutor) [reply homework-2.0/task3.rb]: Is my new solution satisfactory or can it be done more correctly?

**Pull request merged**